lib/Interpreter/IncrementalParser.cpp

```diff
@@ -280,6 +280,55 @@ namespace cling {
     return m_Consumer->getTransaction();
   }
 
+  const Transaction*
+  IncrementalParser::mergeTransactionsAfter(const Transaction *T, bool Prev) {
+    llvm::SmallVector<Transaction*, 8> merge;
+    for (std::deque<Transaction*>::reverse_iterator itr = m_Transactions.rbegin(),
+            end  = m_Transactions.rend(); itr != end; ++itr) {
+      Transaction *cur = *itr;
+      merge.push_back(cur);
+      if (cur==T) {
+        if (!Prev)
+          merge.pop_back();
+        break;
+      }
+    }
```

Comment:
I don't think your comment addresses my concern:

If I call `IncrementalParser::mergeTransactionsAfter((Transaction*)0x123, prev)` it will merge all transactions into the first. Or crash. Probably depending on the value of `prev`. I don't see how `!merge.empty()` prevents that, as `merge` will contain *all* transactions.